homeassistant/components/upnp/binary_sensor.py

```diff
@@ -0,0 +1,134 @@
+"""Support for UPnP/IGD Sensors."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any, Mapping
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_CONNECTIVITY,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    CONFIG_ENTRY_SCAN_INTERVAL,
+    CONFIG_ENTRY_UDN,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    DOMAIN_DEVICES,
+    LOGGER as _LOGGER,
+    UPTIME,
+    WANIP,
+    WANSTATUS,
+)
+from .device import Device
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the UPnP/IGD sensors."""
+    udn = config_entry.data[CONFIG_ENTRY_UDN]
+    device: Device = hass.data[DOMAIN][DOMAIN_DEVICES][udn]
+
+    update_interval_sec = config_entry.options.get(
+        CONFIG_ENTRY_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+    )
+    update_interval = timedelta(seconds=update_interval_sec)
+    _LOGGER.debug("update_interval: %s", update_interval)
+    _LOGGER.debug("Adding sensors")
+    coordinator = DataUpdateCoordinator[Mapping[str, Any]](
+        hass,
+        _LOGGER,
+        name=device.name,
+        update_method=device.async_get_status,
+        update_interval=update_interval,
+    )
+    device.coordinator = coordinator
+
+    await coordinator.async_refresh()
+
+    sensors = [
+        UpnpStatusSensor(coordinator, device),
+    ]
+    async_add_entities(sensors, True)
+
+
+class UpnpStatusSensor(CoordinatorEntity, BinarySensorEntity):
+    """Base class for UPnP/IGD sensors."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator[Mapping[str, Any]],
+        device: Device,
+    ) -> None:
+        """Initialize the base sensor."""
+        super().__init__(coordinator)
+        self._device = device
+
+    @property
+    def icon(self) -> str:
+        """Icon to use in the frontend, if any."""
+        return "mdi:server-network"
```

Comment:
This is static and should be defined as a class attribute with  `_attr_icon = "mdi:server-network"`

Reply:
We don't define icons when we define device_class